### ISSUE TITLE
Non Issue: Cauldron Development.

### ISSUE THREAD

**Issue by Terminal-Access** [gamerforEA/Cauldron-Reloaded#9]
First id like to thank you gameforEA for taking up the work your doing here, and secondly wanted to let you know, if you were not alredy aware, that there is a group of us attempting to do the exact same thing, Bring cauldron back from its untimly demise, were all located over at yive's Mirror http://tcpr.ca/forums/ - this is an open invitation to your self and every one else here to come join us!

hoping that if we can get enough heads all in the one place focusing on bringing cauldron back then we will make progress a lot faster!

@Dark-Arcana - If possible could you achive up any patches, and any thing else cauldron related that you may still have and drop them on a file share some where e.g (mega (dot) co (dot) nz) for me as that would be a huge help!, I current have a huge archive of cauldron related patches and various files though i am probabily missing things. 

thanks

Terminal-Access - Lee

**Comment by Shybella** [comment]
Why jump to another website? It's completely fine here, all your doing is archiving the files....

**Comment by Terminal-Access** [comment]
Acctually were working on rebuilding all of the patches back in to source code, as most releaces (Cauldron-reloaded included) have been build on the 189 sourcecode, so are missing all the patches from that to the last official build "213" and those beyond (up to unoffical build 221 now)

To be clear i am in no way saying jump ship here, please by all means continue on here, i just think if we can have a single place where all those who are working on cauldron can colaborate we all would benefit from it.

**Comment by spannerman79** [comment]
In that case why not perform a pull request @Terminal-Access ?

**Comment by Terminal-Access** [comment]
the idea here wasent just to get new code, its to try and bring the now fragmented Cauldron community back to geathr hence the invite to the forums.

plus Im not 100% sure where this repo stands in the way of what patch/build level its at.

I currently have a copy of the origional repository that has full commits up to build 198 and going all the way back to to build 3.

For Legal reasons i am removing any commits and history prior to the 1.6.4 and gradle commit.
I also in the process of seporating the versions "1.6.4, 1.7.2 and 1.7.10" in to their own respective repos for ease of updating.

The idea here is eventually to bring any missing code/patches back in to the repository and then build on from there for each MC version of Cauldron.

e.g for 1.7.10 we know that were missing sourcode from build 198->204 and then there are official patches (in bps format) from build 205->213.

and as i stated i done know where this repository stands in the way of patch/build level so thats why i have yet to create any code requests.

Hope this clears any confusion up.

**Comment by spannerman79** [comment]
Well by the looks of the readme ( https://github.com/gamerforEA/Cauldron-Reloaded/blob/master/README.md ) pull requests are pretty clear

Just have to wait until @gamerforEA states what build this git repo is based on

And to be honest @Terminal-Access anything that is open sourced or open to the public will eventually become fragmented. Look at Linux which is a perfect example where different people have different ideas of what direction code should go.

**Comment by Shybella** [comment]
"plus Im not 100% sure where this repo stands in the way of what patch/build level its at." does this not address that issue in the long run? blood has been providing patches via his irc now for awhile..

**Comment by Terminal-Access** [comment]
*facepalm* you know what just forget this whole post..

**Comment by Shybella** [comment]
github is the perfect place for this to be.. dcma isnt much of a issue now

**Comment by Terminal-Access** [comment]
no drama was intended, I wasent trying to get this off or move this away from github, so please dont put words in my mouth, and yes im aware of and have all of the current bps patches that blood has given out this far, there one of the things im currently trying to re-work or backwards engineer back in to sourcode so that we can all benefit for it, 

All i was trying to do here was help the community and those working on cauldron get in toutch with eachother, so we all can better colaborate on Cauldron.

**Comment by Shybella** [comment]
if thats what you're trying to do then i'd join the spongepowered community. blood recommended this and is working fluently on the projects.

**Comment by Terminal-Access** [comment]
Sponge is for MC 1.8+ only, i and the community that i support are unable, due to our mods and plugins to move to 1.8 just yet.

**Comment by Shybella** [comment]
no, it'll be back supported to 1.6.4/1.7.10

**Comment by Terminal-Access** [comment]
well as sponge is still in devlopment it may take some time before they begin to back port and have a stable releace out, at that time i may choose to switch, untill sutch time however i am left with Cauldron as my prefered server of choise.

**Comment by Shybella** [comment]
How sponge api and mixin backporting will be much less issue

**Comment by Cracks777** [comment]
sponge wont be done for months  or even a year, they have been working on it for almost a year and still are no where near done

**Comment by Shybella** [comment]
no? sponge has already been implemented ontop of vanilla and has been since feb.. rev 2.0 came out a few days ago which is basically 90% of the api finished, mixin and the sponge mod for forge is on the same state.

**Comment by jburrows82** [comment]
Where is there official disclosure about their intentions of backporting to 1.6.4? The backport to 1.7 was stated in the current state address that happened in Feb, as such to be a secondary priority AFTER a stable 1.8 implementation. 

Cauldron is very much a huge priority to the modding community that mostly has openly declared staying on 1.7 and NOT going to 1.8, which puts Sponge as a secondary option until backporting is complete.

**Comment by Shybella** [comment]
if you followed blood and are on his irc you'd know this.

**Comment by jburrows82** [comment]
Ahh IRC gossip. Good to know, thank you for confirming.

**Comment by Shybella** [comment]
gossip? its his patron irc lol... not the main room this whole issue is a joke lol

**Comment by Myrninvollo** [comment]
Also as stated last night in state of sponge 3  there will be no official back port to 1.7.10 of sponge.  as for me and my team were sticking with cauldron, since we done to many edits ( rpg related) . And I'm not re writing code again like i did from mcpc > mcpc+ > cauldron.  and. We thank gamerforea for cualdron reloaded.

**Comment by Shybella** [comment]
https://forums.spongepowered.org/t/the-complete-beginners-guide-to-sponge-for-server-owners-and-other-non-coders/4345

**Comment by Myrninvollo** [comment]
Shybella,  thus is about cauldron not sponge. So unless u have something related to cauldron,  plz move back to the sponge forums. Thank u

**Comment by Shybella** [comment]
I'm only correcting you. This whole issue has absolutely nothing to do with Cauldron for productivity. It's even more asinine that we've got people thinking that archiving binary files for public download will some how bring together a community and also negating this fork of Cauldron on a well-known open source influenced community.

What's even more disgusting more than half these patches can be found under Spigot's Stashhub, which is all all sourced. You want the source for the BPS is easy. You can easily use a decomplier on the complied jar classes to find what classes we're modified and decompile those and compare from the complied project with the binary patches applied. 

As for the state of Cauldron. Nothing is really wrong with the project and yes, give a high-five I guess for updating Forge. He's gone out of his way and took the time to provide this properly without looking for credit and some sort of justification, so I'll agree with you.

**Comment by jburrows82** [comment]
@Myrninvollo You're right, Lex and crew have convinced the rest of the team that even the original secondary goal of backporting to 1.7 won't be happening by them. That or the sponge team didn't meet up on IRC to find out what's really going on first, perhaps? 

Official source:
https://forums.spongepowered.org/t/state-of-sponge-iii-april-2015/6813/5

**Comment by Myrninvollo** [comment]
Thank you , @jburrows82

**Comment by TheBrain0110** [comment]
Ok, since things have gotten off track, I see what the point is here.

There's been a series of unofficial Cauldron patches released ever since the official patches from blood stopped around b212. I've personally been running 216 (Forge 1307) for a while. They now appear to be up to b221, on Forge 1381.

I don't know who's been developing them, or where, and what other patches have been included other than Forge updates.

Then we have this repository, which we can at least see, but don't know which base build it was forked from, so don't know which updates have been included.

So I think the point was to get the multiple groups doing underground maintenance talking together, and it would seem to make sense to have the people behind Yive's mirror contribute to this repo so they can all be on the same page and we don't have multiple different sources.

**Comment by Sayshal** [comment]
TheBrain0110 where did you get b221? I'm running Forge 1381 with many issues.

**Comment by TheBrain0110** [comment]
It's available at Yive's mirror, and others have said it runs well for them, but I haven't actually tried that one yet. I'm still on b216-1307, trying to decide which one to move to.

**Comment by Sayshal** [comment]
Well I'm having issues getting this whole thing to work on TPPI@ v0.1.4
(which requires 1384 or higher I believe), the 1388 one is not liking
Mystcraft at all.

**Comment by Myrninvollo** [comment]
i agree with @TheBrain0110 on 1 thing , that every single cauldron 3rd party teams needs to push there patches here. simple fact of the matter is this . the mc modding community is now split in to 2 groups 1.7.10 and 1.8+. now the 1.7.10 group is staying becouse they rather add more content and imporvements to there mods/plugins, the 1.8+ group is mostlsy new Modders and some old modders that are in my opion Rightwing with Forge. now as for most server owners they just want to update there servers to the lasest mc Version to pull in new players (loaded with money and becouse the will break the elua no matter what) . MC only updates with new stuff to pull in new Players. 3 things wil soon happend. 1)MC stops gettting updated becouse they ran out of new fetures (most wont even make to the consoles) or 2) the Offical API gets public relase (Cough) or 3) mc modding dies becouse mod/plugin devs leave becouse they fed up with updateing.

**Comment by spannerman79** [comment]
I couldn't help but laugh at some of your points @Myrninvollo .
Point 1) Yeap! I agree although there are two sort of server groups - first one doesn't care about ripping off players and break any EULA or mod terms. And there is the second that only want to cover their costs and no more.

Point 2) I think the Mojang API is vapour ware - it won't ever come and if it does it will be like the so called comeback of Duke Nukem. Broken, poorly implemented and focused (i suspect) on just 1.8

Pont 3) MC modding won't die fully - there are still devs working on their plugins and mods for 1.7.10.  Actually 1.7.10 have the biggest list of mods from what I can gather ( http://bot.notenoughmods.com/ )

**Comment by Myrninvollo** [comment]
@spannerman79  well at least it made you laugh , all we need to do now is find out which mod/plugin devs aren't moving to 1.8 and get them to come and help fix any bugs between there mods and cauldron  :)

**Comment by spannerman79** [comment]
Agreed @Myrninvollo .  But where exactly?
Personally I don't trust the OP as (to me) it sounded more like a sales pitch.  Perhaps it was worded incorrectly or didn't have the intent it had but (again to me) has left a sour taste in the mouth so to speak.

I don't trust Curse with this either as they have their heads so far up Mojang/Microsoft's back side it isn't funny.  Now that Mojang is under the control of Microsoft (i don't care what so called news articles state - Microsoft bought it - they own it.  Fairly simple to understand that) I can expect soon to be Windows only "features" for Minecraft "2.0". Say goodbye to good servers hosting Cauldron as - in my opinion - best servers to host is on linux

But here is the thing - cost.  The cost involved to run a system on the scale on *just* 1.7.10 minecraft plus cauldron would be a pretty daunting task alone.
I know build servers are not cheap - as that would be the only incentive to bring over devs is to offer a build server or share a build server (Jenkins) with others.

The location of where the server or servers hosted would be a factor too.  Yes currently USA based servers offer the best price for storage/bandwidth per $ but all that is needed is a simple take down notice - doesn't have to even be legit - especially if its from Microsoft and say bye bye to data on that server infrastructure.

Then there are the server owners that use systems like solder (solder.io) that need a direct link to the mod in question - so out goes adf.ly and similiar cash link systems for the devs to make some sort of extra income.

**Comment by TheBrain0110** [comment]
Ok, you guys are getting way off topic again. This isn't about the Mojang API, or theoretical Microsoft takedowns in the future, or who hosts what, or even the split among mod and Forge developers between 1.7 / 1.8, and Forge vs Cauldron vs Sponge.

This is about the multiple groups just maintaining Cauldron 1.7. There's at least this Github project, and the unknown people publishing builds on Yive's Mirror, and possibly other sources. All we care about is making sure that the patches between all those are merged, so we can have the best Cauldron builds for 1.7, until the prolonged split with the mod developers is resolved and Sponge is ready, and we can move our servers to 1.8 and have most of the mods we use ready there.

**Comment by Myrninvollo** [comment]
@TheBrain0110 "This is about the multiple groups just maintaining Cauldron 1.7. There's at least this Github project, and the unknown people publishing builds on Yive's Mirror, and possibly other sources. All we care about is making sure that the patches between all those are merged, so we can have the best Cauldron builds for 1.7" , i agree


and @spannerman79 we have a host that host all versions of MC including Cauldron. 

and yet again i see from these post just why the word Commuinty has left MC. well i wish every one luck in what you do, as for me and my group  we aint moving  to 1.8 becouse our project thats took 3 years in the making aint moveing of 1.7.10 even if that means were the only group left with cauldron.
